src/vmspawn/vmspawn.c

```diff
@@ -595,6 +598,32 @@ static int start_tpm(sd_bus *bus, const char *scope, const char *our_runtime_dir
         return 0;
 }
 
+static int finalize_root(char **ret) {
+        int r;
+        _cleanup_(dissected_image_unrefp) DissectedImage *image = NULL;
+        _cleanup_free_ char *root = NULL;
+
+        assert(ret);
+
+        r = dissect_image_file_and_warn(arg_image, NULL, NULL, NULL, 0, &image);
```

Comment:
we generally want that parameters in func calls where one has no idea what they could be because they are nullified are prefixed with param names as comments. hence please rewrite as:

```c
r = dissect_image_file_and_warn(
                arg_image,
                /* verity= */ NULL,
                /* mount_options= */ NULL,
                /* image_policy= */ NULL,
                /* flags= */ 0,
                &image);
```

Reply:
i figure you want to specify DISSECT_IMAGE_REQUIRE_ROOT here.